SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityTaskReconciliation.java

```diff
@@ -151,6 +160,8 @@ private void checkReconciliation(final SchedulerDriver driver, final long reconc
     if (taskStatuses.isEmpty()) {
       LOG.info("Task reconciliation ended after {} checks and {}", numTimes, JavaUtils.duration(reconciliationStart));
 
+      stateManager.saveTaskReconciliationStatistics(new SingularityTaskReconciliationStatistics(reconciliationStart, System.currentTimeMillis() - reconciliationStart, numTimes, remainingTaskCounts));
```

Comment:
this duration (` System.currentTimeMillis() - reconciliationStart`) is not actually an indicator of how long task reconciliation has taken. Since the poller runs 30s later, the time reported for reconciliation will always be a multiple of 30s (or the configured wait time). The time we actually need to mark the finish is the most recent statusUpdate whose reason was `REASON_RECONCILIATON`